plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/encryption/EncryptionHeadersTranslator.java

```diff
@@ -35,14 +36,40 @@ static EncryptionHeadersTranslator encryptionHeadersTranslator(Location location
                 .orElseThrow(() -> new IllegalArgumentException("Unknown location scheme: " + location));
     }
 
-    private static EncryptionHeadersTranslator forScheme(String scheme)
+    static EncryptionHeadersTranslator forScheme(String scheme)
     {
         // These should match schemes supported in the FileSystemSpoolingModule
-        return switch (scheme) {
+        EncryptionHeadersTranslator schemeHeadersTranslator = switch (scheme) {
             case "s3" -> new S3EncryptionHeadersTranslator();
             case "gs" -> new GcsEncryptionHeadersTranslator();
             case "abfs" -> new AzureEncryptionHeadersTranslator();
             default -> throw new IllegalArgumentException("Unknown file system scheme: " + scheme);
         };
+
+        // Normalize header case so it won't matter which case we will get from the client
+        return new NormalizingHeadersTranslator(schemeHeadersTranslator);
+    }
+
+    class NormalizingHeadersTranslator
```

Comment:
`@VisibleForTesting`?

Reply:
It's not visible for testing :) It's applied automatically when you retrieve header translator for given scheme